src/gui/userstatus.cpp

```diff
@@ -26,45 +26,39 @@
 
 namespace OCC {
 
-UserStatus::UserStatus(AccountState *accountState, QObject *parent)
+UserStatus::UserStatus(QObject *parent)
     : QObject(parent)
-    , _accountState(accountState)
     , _status("online")
     , _message("")
 {
-    connect(this, &UserStatus::fetchedCurrentUserStatus, _accountState, &AccountState::userStatusChanged);
+
 }
 
-void UserStatus::fetchStatus()
+void UserStatus::fetchUserStatus(AccountStatePtr account)
 {
     if (_job) {
         _job->deleteLater();
     }
 
-    AccountPtr account = _accountState->account();
-    _job = new JsonApiJob(account, QStringLiteral("/ocs/v2.php/apps/user_status/api/v1/user_status"), this);
-    connect(_job.data(), &JsonApiJob::jsonReceived, this, &UserStatus::slotFetchedCurrentStatus);
+    _job = new JsonApiJob(account.data()->account(), QStringLiteral("/ocs/v2.php/apps/user_status/api/v1/user_status"), this);
+    connect(_job.data(), &JsonApiJob::jsonReceived, this, &UserStatus::slotFetchUserStatusFinished);
+    connect(this, &UserStatus::fetchUserStatusFinished, account.data(), &AccountState::userStatusChanged);
```

Comment:
@camilasan I'd prefer connecting this signal to a slot in the `AccountState::fetchUserStatus()`. I guess this is how we usually do this, and it would look more readable and is safer as well. This is why we have this signal after all `UserStatus::fetchUserStatusFinished` :) So, it looks good if we do:

```
void AccountState::fetchUserStatus() {
  connect(_userStatus, &UserStatus::fetchUserStatusFinished, this, &AccountState::userStatusChanged);
    _userStatus->fetchUserStatus(AccountStatePtr(this));
}
```
It's also kinda unusual to see one class subscribing another class to a signal, without the former even knowing about it.

Reply:
Yes, good point.

Reply:
> It's also kinda unusual to see one class subscribing another class to a signal, without the former even knowing about it.

Not at all! This is the whole point of signals/slots and connect being designed that way! Any time you connect say a QSpinBox to a QSlider you're exactly in the case of a third party connecting two instances of two classes which don't know each other.